lib/internal/modules/cjs/loader.js

```diff
@@ -369,10 +370,10 @@ function resolveExports(nmPath, request, absoluteRequest) {
       if (dirMatch !== '') {
         const dirMapping = pkgExports[dirMatch];
         const remainder = StringPrototype.slice(mappingKey, dirMatch.length);
-        const expectedPrefix = path.resolve(basePath, dirMapping);
-        const resolved = path.resolve(expectedPrefix, remainder);
+        const expectedPrefix = new URL(dirMapping, baseURL).href;
```

Comment:
Minor nit - maybe leave it as the URL object here, and instead use `expectedPrefix.href` below. In theory a URL object as the referrer should have better perf.

Reply:
Updated! See latest `--fixup` commit.